src/metabase/models/setting.clj

```diff
@@ -615,12 +616,10 @@
   if any."
   [setting-definition-or-name]
   (let [{:keys [cache? getter enabled? default]} (resolve-setting setting-definition-or-name)
-        disable-cache?                           (not cache?)]
+        disable-cache?                           (or *disable-cache* (not cache?))]
     (if (or (nil? enabled?) (enabled?))
-      (if (= *disable-cache* disable-cache?)
-        (getter)
-        (binding [*disable-cache* disable-cache?]
-          (getter)))
+      (binding [*disable-cache* disable-cache?]
+        (getter))
```

Comment:
There was actually a bug here: if `*disable-cache*` was truthy it would get stomped on if the Setting in question was a normal cached Setting. Fixed.

Also the "optimization" to avoid `binding` if the value wouldn't change was super duper silly, it saved about a tenth of a microsecond so really not worth making the code more confusing.

Reply:
```
unoptimized

Evaluation count : 165768 in 6 samples of 27628 calls.
             Execution time mean : 3.758099 µs
    Execution time std-deviation : 346.067313 ns
   Execution time lower quantile : 3.602993 µs ( 2.5%)
   Execution time upper quantile : 4.356128 µs (97.5%)
                   Overhead used : 1.566506 ns

Found 1 outliers in 6 samples (16.6667 %)
	low-severe	 1 (16.6667 %)
 Variance from outliers : 15.8136 % Variance is moderately inflated by outliers

"optimized"

Evaluation count : 173400 in 6 samples of 28900 calls.
             Execution time mean : 3.696985 µs
    Execution time std-deviation : 146.793030 ns
   Execution time lower quantile : 3.590312 µs ( 2.5%)
   Execution time upper quantile : 3.915156 µs (97.5%)
                   Overhead used : 1.566506 ns
```

Reply:
I have https://github.com/hugoduncan/criterium/ in my user `deps.edn`